internal/p2p/channel.go

```diff
@@ -179,18 +193,25 @@ func (iter *ChannelIterator) Envelope() *Envelope { return iter.current }
 //
 // This allows the caller to consume messages from multiple channels
 // without needing to manage the concurrency separately.
-func MergedChannelIterator(ctx context.Context, chs ...*Channel) *ChannelIterator {
+func MergedChannelIterator(ctx context.Context, chs ...Channel) *ChannelIterator {
 	iter := &ChannelIterator{
 		pipe: make(chan Envelope), // unbuffered
 	}
 	wg := new(sync.WaitGroup)
 
 	for _, ch := range chs {
 		wg.Add(1)
-		go func(ch *Channel) {
+		go func(ch Channel, pipe chan Envelope) {
```

Comment:
```suggestion
		go func(ch Channel, pipe chan<- Envelope) {
```

Do we need to close this when the goroutine exits?

Reply:
we never have closed this channel before, and this just sort of moves the function around. I think it's fine.